.gitlab-ci-check-docker-build.yml

```diff
@@ -78,10 +78,18 @@ stages:
   - publish
 
 .docker_login_registries: &docker_login_registries |
-  docker login -u "$REGISTRY_MENDER_IO_USERNAME" -p "$REGISTRY_MENDER_IO_PASSWORD" registry.mender.io || \
-    echo "Warning: registry.mender.io credentials unavailable or invalid"
-  docker login -u "$DOCKER_HUB_USERNAME" -p "$DOCKER_HUB_PASSWORD" || \
-    echo "Warning: Docker credentials unavailable or invalid"
+  if [ -n "$REGISTRY_MENDER_IO_USERNAME" ]; then \
```

Comment:
I think you should check for the PASSWORD variables instead, which is the variable that for sure will be protected and potentially not available.

Reply:
Nice trick!